tests/e2e/ctl_v3_member_no_proxy_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !cluster_proxy
+
+package e2e
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"go.etcd.io/etcd/server/v3/etcdserver"
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+	"go.etcd.io/etcd/tests/v3/framework/testutils"
+)
+
+func TestMemberReplaceMultiple(t *testing.T) {
+	// Loop for adding member is hard to repro
+	for i := 0; i < 10; i++ {
+		succeed := t.Run("LoopForAdd", func(t *testing.T) {
+			memberReplaceTest(t, true)
+		})
+		if !succeed {
+			t.Fatalf("LoopForAdd failed")
+		}
+	}
+	// Only call member add once is easy to repro
+	for i := 0; i < 10; i++ {
+		succeed := t.Run("OneStepAdd", func(t *testing.T) {
+			memberReplaceTest(t, false)
+		})
+		if !succeed {
+			t.Fatalf("LoopForAdd failed")
+		}
+	}
+}
+
+func memberReplaceTest(t *testing.T, loopForAdd bool) {
+	e2e.BeforeTest(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	epc, err := e2e.NewEtcdProcessCluster(ctx, t)
+	require.NoError(t, err)
+	defer epc.Close()
+
+	memberId := rand.Int() % len(epc.Procs)
+	member := epc.Procs[memberId]
+	memberName := member.Config().Name
+	var endpoints []string
+	for i := 1; i < len(epc.Procs); i++ {
+		endpoints = append(endpoints, epc.Procs[(memberId+i)%len(epc.Procs)].EndpointsGRPC()...)
+	}
+	cc, err := e2e.NewEtcdctl(epc.Cfg.Client, endpoints)
+	require.NoError(t, err)
+
+	c := epc.Etcdctl()
+	memberID, found, err := getMemberIdByName(ctx, c, memberName)
+	require.NoError(t, err)
+	require.Equal(t, found, true, "Member not found")
+
+	// Need to wait health interval for cluster to accept member changes
+	time.Sleep(etcdserver.HealthInterval)
+
+	t.Logf("Removing member %s", memberName)
+	_, err = c.MemberRemove(ctx, memberID)
```

Comment:
If the client selects that member which is going to be removed, it might return error because that server stops. 

client won't retry if the error is `code = Unavailable`

Reply:
I follow the guide in https://etcd.io/docs/v3.5/tutorials/how-to-deal-with-membership/ to remove member. Which saying is correct?

Reply:
I run into this issue in my local one time. https://github.com/etcd-io/etcd/blob/dfdffe48f9e7c622f1b863d013754c2a824f6d35/server/etcdserver/server.go#L1115

1 second might good enough to return response. never mind.

Reply:
> If the client selects that member which is going to be removed, it might return error because that server stops.

Yes, actually there are two "issues" (one real issue, one improvement) in the test cases:
- [Issue] The client might connect to the member which will be removed, so the client might get an error response.
- [Improvement] We should use client sdk instead of etcdctl to communicate with etcdserver directly. Otherwise when the client failed, it can only see etcdctl 's log instead of etcdserver's log.

Actually I already updated the # 1 above.  Let me deliver a PR to fix it.

Reply:
FYI. add "EXPECT_DEBUG=true" can get the detailed etcdserver's log, 

```
EXPECT_DEBUG=true go test -run TestMemberReplaceMultiple -v
```